src/components/audio-params.js

```diff
@@ -173,39 +177,44 @@ AFRAME.registerComponent("audio-params", {
 
   update() {
     if (this.audioRef) {
-      this.audioRef.setDistanceModel(this.data.distanceModel);
-      this.audioRef.setRolloffFactor(this.data.rolloffFactor);
-      this.audioRef.setRefDistance(this.data.refDistance);
-      this.audioRef.setMaxDistance(this.data.maxDistance);
-      this.audioRef.panner.coneInnerAngle = this.data.coneInnerAngle;
-      this.audioRef.panner.coneOuterAngle = this.data.coneOuterAngle;
-      this.audioRef.panner.coneOuterGain = this.data.coneOuterGain;
+      if (this.data.audioType === AudioType.PannerNode) {
+        this.audioRef.setDistanceModel(this.data.distanceModel);
+        this.audioRef.setRolloffFactor(this.data.rolloffFactor);
+        this.audioRef.setRefDistance(this.data.refDistance);
+        this.audioRef.setMaxDistance(this.data.maxDistance);
+        this.audioRef.panner.coneInnerAngle = this.data.coneInnerAngle;
+        this.audioRef.panner.coneOuterAngle = this.data.coneOuterAngle;
+        this.audioRef.panner.coneOuterGain = this.data.coneOuterGain;
+      }
       this.data.gain !== undefined && this.updateGain(this.data.gain);
```

Comment:
The `updateGain` function should not write to `this.data.gain`. (It should also not call `setAttribute` because then we would introduce an infinite loop here.) Instead, `updateGain` should apply the derived gain value to the underlying node, even if that differs from `this.data.gain`. If it is important for `this.data.gain` to be exactly equal to the gain of the underlying audio node, then whenever `update` receives an invalid gain node, it should call `setAttribute` to force it back to a valid value.

Reply:
Updated